### PR TITLE
fix(dashboard): update trading coin explicitly

### PR DESCRIPTION
**Pull request by UncleBill**
## Description

Closes [MASK-279](https://mask.atlassian.net/browse/MASK-279)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/dashboard/src/pages/Wallets/components/FungibleTokenTable/index.tsx [hunk 1]
```diff
@@ -95,10 +95,10 @@ export const FungibleTokenTable = memo<TokenTableProps>(({ selectedChainId }) =>
         openSwapDialog({
             open: true,
             traderProps: {
-                coin: {
+                defaultFromCoin: {
                     id: token.id,
-                    name: token.name ?? '',
-                    symbol: token.symbol ?? '',
+                    name: token.name || '',
```
Comment (UncleBill) [inline packages/dashboard/src/pages/Wallets/components/FungibleTokenTable/index.tsx:100]: This is an abuse of `??`, what if token.name is `0` at runtime?

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx [hunk 2]
```diff
@@ -89,6 +91,25 @@ export function Trader(props: TraderProps) {
     }, [chainId, chainIdValid])
     //#endregion
 
+    const updateTradingCoin = useCallback(
+        (type: any, coin?: Coin) => {
+            if (!coin) return
```
Comment (hanszhoou) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx]:
```suggestion
            if (!coin?.contract_address) return
```

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx [hunk 3]
```diff
@@ -89,6 +91,25 @@ export function Trader(props: TraderProps) {
...
+            dispatchTradeStore({
+                type,
+                token: coin.contract_address
+                    ? createERC20Token(chainId, coin.contract_address, decimals, coin.name, coin.symbol)
+                    : undefined,
```
Comment (hanszhoou) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx]:
```suggestion
                token: createERC20Token(chainId, coin.contract_address, decimals, coin.name, coin.symbol),
```

**Review by hanszhoou — DISMISSED** [review]

File: packages/dashboard/src/pages/Wallets/components/FungibleTokenTable/index.tsx (hunk 1 above)
Comment (guanbinrui) [reply packages/dashboard/src/pages/Wallets/components/FungibleTokenTable/index.tsx:100]: Why a token's name will be `0`?

Comment (UncleBill) [reply packages/dashboard/src/pages/Wallets/components/FungibleTokenTable/index.tsx:100]: > Why a token's name will be `0`?

Just as an example. And well, runtime can be unpredictable.

**Comment by guanbinrui** [comment]
@DimensionDev/qa

**Review by Jack-Works — DISMISSED** [review]

**Comment by AinuX** [comment]
Checked

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**